lib/utility/src/WatchdogWrapper.cpp

```diff
@@ -0,0 +1,34 @@
+
+#include "WatchdogWrapper.h"
+
+#include "Logger.h"
+#include "ResetReason.h"
+#include "mbed.h"
+
+namespace Utility {
+
+Thread WatchdogWrapper::pet_thread;
+
+void WatchdogWrapper::startWatchdog(std::chrono::milliseconds countdown_ms /*= 5000ms*/,
+                                    std::chrono::milliseconds pet_ms /*= 1000ms*/) {
+  uint32_t countdown_uint32 = countdown_ms.count();
+  Watchdog &watchdog        = Watchdog::get_instance();
+  watchdog.start(countdown_uint32);
+  pet_thread.start(callback(WatchdogWrapper::petWatchdog, &pet_ms));
+}
+
+void WatchdogWrapper::logResetReason() {
+  const reset_reason_t reason = ResetReason::get();
+  if (reason == RESET_REASON_WATCHDOG) {
+    time_t seconds = time(NULL);
+    Utility::logger << "Reset Reason: Watchog Timer - " << ctime(&seconds) << "\r\n";
```

Comment:
Suggestion: change this to something like `"Reset Reason: Watchdog Timer - Timestamp: " << seconds " seconds since power on\r\n"`